src/plugin/browser_field.rs

```diff
@@ -3,12 +3,16 @@ use crate::{
     PathKind, Plugin, ResolveResult, Resolver, State,
 };
 use path_absolutize::Absolutize;
-use std::path::Path;
+use std::path::{Path, PathBuf};
 
 pub struct BrowserFieldPlugin<'a> {
     pkg_info: &'a DescriptionData,
 }
 
+fn append_ext_for_path(path: &Path, ext: &str) -> PathBuf {
+    PathBuf::from(&format!("{}{ext}", path.display()))
```

Comment:
Shouldn't this be something similar to https://doc.rust-lang.org/stable/std/path/struct.Path.html#method.with_extension?

`with_extension(path.display().trim_start('.')`. This reduces allocation, and does a better job at allocations https://doc.rust-lang.org/stable/src/std/path.rs.html#1488

Reply:
We can't use this because its handling of `'.'` doesn't handle it the way we expect it to

Reply:
You can mimic how it's done in  https://doc.rust-lang.org/stable/src/std/path.rs.html#1488 so you don't need to allocate twice: once in `format` and once in `PathBuf::from`